api/src/opentrons/hardware_control/backends/tip_presence_manager.py

```diff
@@ -0,0 +1,95 @@
+import logging
+from functools import partial
+from typing import Callable, Optional, List, Dict
+
+
+from opentrons.hardware_control.types import TipStateType, OT3Mount
+
+from opentrons_hardware.drivers.can_bus import CanMessenger
+from opentrons_hardware.firmware_bindings.constants import NodeId
+from opentrons_hardware.hardware_control.tip_presence import (
+    TipDetector,
+    types as tip_types,
+)
+from opentrons_shared_data.errors.exceptions import UnmatchedTipPresenceStates
+
+log = logging.getLogger(__name__)
+
+TipListener = Callable[[OT3Mount, bool], None]
+TipDetectorByMount = Dict[OT3Mount, Optional[TipDetector]]
+TipUpdateByMount = Dict[OT3Mount, Optional[bool]]
+
+
+class TipDetectorNotFoundError(Exception):
+    pass
+
+
+def _mount_to_node(mount: OT3Mount) -> NodeId:
+    return {
+        OT3Mount.LEFT: NodeId.pipette_left,
+        OT3Mount.RIGHT: NodeId.pipette_right,
+    }[mount]
+
+
+class TipPresenceManager:
+    """Handle tip change notification coming from CAN."""
+
+    _listeners: List[TipListener]
+    _detectors: TipDetectorByMount
+    _last_state: TipUpdateByMount
+
+    def __init__(self, can_messenger: CanMessenger) -> None:
+        self._messenger = can_messenger
+        self._listeners = []
+        self._detectors = {m: None for m in OT3Mount}
+        self._last_state = {m: None for m in OT3Mount}
+
+    async def build_detector(self, mount: OT3Mount, sensor_count: int) -> None:
+        # clear detector if pipette does not exist
+        if not sensor_count > 0:
```

Comment:
```suggestion
        if sensor_count <= 0:
```
nit, makes it easier to read for me